packages/components/select/src/select.vue

```diff
@@ -365,6 +366,30 @@ export default defineComponent({
     const API = useSelect(_props, emit)
     const { calculatorRef, inputStyle } = useCalcInputWidth()
 
+    onBeforeMount(() => {
+      if (!props.persistent) {
+        // After option rendering is completed, the useSelect internal state can collect the value of each option.
+        // If the persistent value is false, option will not be rendered by default, so in this case,
+        // manually render and load option data here.
+        const hasValue = isArray(modelValue.value)
+          ? modelValue.value.length > 0
+          : !!modelValue.value
+        if (hasValue) {
```

Comment:
It's unnecessary to judge the value of `hasValue`. For example, the initial value was empty, then the value changed. The label can't be shown correctly.

Reply:
That makes sense. 👍